src/lightning/fabric/accelerators/tpu.py

```diff
@@ -68,9 +75,14 @@ def get_parallel_devices(devices: Union[int, List[int]]) -> List[torch.device]:
     def auto_device_count() -> int:
         """Get the devices when set to auto."""
         import torch_xla.core.xla_env_vars as xenv
+        from torch_xla.experimental import pjrt, tpu
         from torch_xla.utils.utils import getenv_as
 
-        return getenv_as(xenv.TPU_NUM_DEVICES, int, 8)
+        if pjrt.using_pjrt():
+            device_count_on_version = {2: 8, 3: 8, 4: 4}
+            return device_count_on_version.get(tpu.version(), 8)
```

Comment:
PJRT don't set up or rely on the `TPU_NUM_DEVICES` env var.

Reply:
Shouldn't this get the topology and multiply it by 2 in the case of v4? https://cloud.google.com/tpu/docs/system-architecture-tpu-vm

Reply:
PyTorch/XLA treats a V4 chip as a single logical device, so there are 4 devices on a V4 node.

Reply:
But doesn't it depend on whether it's a `v4-8` or `v4-128`?

As you said, "v4 chip == logical device" and v4-8 has 4 chips but v4-128 64

Sorry if I'm missing something obvious!

Reply:
I saw that `pjrt.local_process_count()` is also available

Reply:
>  As you said, "v4 chip == logical device" and v4-8 has 4 chips but v4-128 64

Here I think only the local count matters, it is essentially number of devices per host.

Reply:
> I saw that `pjrt.local_process_count()` is also available

That's available only after `xmp.spawn()`